contracts/BC_fusion/StakeHub.sol

```diff
@@ -0,0 +1,1203 @@
+        emit Redelegated(srcValidator, dstValidator, delegator, shares, newShares, bnbAmount);
+
+        address[] memory stakeCredits = new address[](2);
+        stakeCredits[0] = srcValInfo.creditContract;
+        stakeCredits[1] = dstValInfo.creditContract;
+        IGovToken(GOV_TOKEN_ADDR).syncBatch(stakeCredits, delegator);
+        if (delegateVotePower) {
+            IGovToken(GOV_TOKEN_ADDR).delegateVote(delegator, dstValidator);
+        }
+    }
+
+    /**
+     * @dev Claim the undelegated BNB from the pool after unbondPeriod
+     * @param operatorAddress the operator address of the validator
+     * @param requestNumber the request number of the undelegation. 0 means claim all
+     */
+    function claim(address operatorAddress, uint256 requestNumber) external whenNotPaused notInBlackList {
+        _claim(operatorAddress, requestNumber);
+    }
+
+    /**
+     * @dev Claim the undelegated BNB from the pools after unbondPeriod
+     * @param operatorAddresses the operator addresses of the validator
+     * @param requestNumbers numbers of the undelegation requests. 0 means claim all
+     */
+    function claimBatch(
+        address[] calldata operatorAddresses,
+        uint256[] calldata requestNumbers
+    ) external whenNotPaused notInBlackList {
+        if (operatorAddresses.length != requestNumbers.length) revert InvalidRequest();
+        for (uint256 i; i < operatorAddresses.length; ++i) {
+            _claim(operatorAddresses[i], requestNumbers[i]);
+        }
+    }
+
+    /**
+     * @dev Sync the gov tokens of validators in operatorAddresses
+     * @param operatorAddresses the operator addresses of the validators
+     * @param account the account to sync gov tokens to
+     */
+    function syncGovToken(
+        address[] calldata operatorAddresses,
+        address account
+    ) external whenNotPaused notInBlackList {
+        uint256 _length = operatorAddresses.length;
+        address[] memory stakeCredits = new address[](_length);
+        address credit;
+        for (uint256 i = 0; i < _length; ++i) {
+            if (!_validatorSet.contains(operatorAddresses[i])) revert ValidatorNotExist(); // should never happen
+            credit = _validators[operatorAddresses[i]].creditContract;
+            stakeCredits[i] = credit;
+        }
+
+        IGovToken(GOV_TOKEN_ADDR).syncBatch(stakeCredits, account);
+    }
+
+    /*----------------- system functions -----------------*/
+    /**
+     * @dev This function will be called by consensus engine. So it should never revert.
+     */
+    function distributeReward(address consensusAddress) external payable onlyValidatorContract {
+        address operatorAddress = consensusToOperator[consensusAddress];
+        Validator memory valInfo = _validators[operatorAddress];
+        if (valInfo.creditContract == address(0) || valInfo.jailed) {
+            emit RewardDistributeFailed(operatorAddress, "INVALID_VALIDATOR");
+            return;
```

Comment:
How do we deal with the funds that stuck in this contract?

Reply:
What about send to systemReward?